config/config.go

```diff
@@ -1043,13 +1032,7 @@ func SetServerDefaults(v *viper.Viper) error {
 	if IsRootExecution() {
 		v.SetDefault(param.Origin_RunLocation.GetName(), filepath.Join("/run", "pelican", "xrootd", "origin"))
 		v.SetDefault(param.Cache_RunLocation.GetName(), filepath.Join("/run", "pelican", "xrootd", "cache"))
-
-		// Several deprecated keys point to Cache.StorageLocation, and by the time we reach this section of code, we should
-		// have already mapped those keys in handleDeprecatedConfig(). To prevent overriding potentially-mapped deprecated keys,
-		// we only re-set he default here if this key is not set.
-		if !v.IsSet(param.Cache_StorageLocation.GetName()) {
-			v.SetDefault(param.Cache_StorageLocation.GetName(), filepath.Join("/run", "pelican", "cache"))
-		}
+		v.SetDefault(param.Cache_StorageLocation.GetName(), filepath.Join("/run", "pelican", "cache"))
```

Comment:
Hmm, there's a bigger issue here as well, and this section of code has now broken. The issue is that the parameters `Cache.NamespaceLocation`, `Cache.DataLocations` and `Cache.MetaLocations` are supposed to be scoped under `Cache.StorageLocation`, if the value is set.

Moving the `handleDeprecatedConfig` to run after all these defaults are set breaks this mapping, because these secondary values inherit the default `Cache.StorageLocation` and are unaware that `Cache.DataLocation` is being used as the value (because that hasn't happened yet).

I think this means we need to re-evaluate the approach and that we should also have a test for this inheritance to make sure it doesn't break.